lib/portage/package/ebuild/fetch.py

```diff
@@ -408,23 +408,22 @@ def get_mirror_url(mirror_url, filename, cache_path=None):
 	if ts >= time.time() - 86400:
 		mirror_conf.deserialize(data)
 	else:
+		tmpfile = '.layout.conf.%s' % urlparse(mirror_url).hostname
 		try:
-			f = urlopen(mirror_url + '/distfiles/layout.conf')
-			try:
-				data = io.StringIO(f.read().decode('utf8'))
-			finally:
-				f.close()
-
-			mirror_conf.read_from_file(data)
+			if fetch({tmpfile: (mirror_url + '/distfiles/layout.conf',)},
+					mysettings, fetchonly=1, try_mirrors=0):
```

Comment:
We need to add a fetch parameter to make it fetch and atomically replace the file even if it happens to exist already.

Reply:
Also, the fetchonly parameter triggers non-blocking mode, but we really need to block here in order to ensure that the file is available when the function returns. In 491b3ff4d35194765c8ee93c6a438097551fd07e I've added a demonstration of the behavior for stale files when no digests are given. I'm going to work on adding a parameter to force fetch now.

Reply:
https://github.com/gentoo/portage/pull/470 implements a suitable fetch force parameter.

Reply:
Thanks! Updated now.